File: problem7/problem7.c
#include <stdio.h>
#include <math.h>

int isprime(int a) {
 	if (a == 2) return 1;
 	if (!(a % 2)) return 0;
 	for (int i = 3; i <= (int)sqrt(a); i += 2)
		if (!(a % i))
			return 0;
	return 1;
}

int main() {
	int n = 1, count = 0;
	while (count < 10000) {
		n += 2;
		if (isprime(n)) count++;
	}
	
	printf("10001st prime number is: %i\n", n);
	return 0;
}

File: problem7/problem7.py
from math import sqrt

def isprime(n):
	if n==2: return True
	if not n%2: return False
	
	for i in range(3, int(sqrt(n))+1, 2):
		if not n % i: return False
	return True

def nth_prime(mx):
	count = 1 #1 because we exclude 2 from the count
	candidate = 1
	while count < mx:
		candidate+=2
		if isprime(candidate): count += 1	

	return candidate

if __name__ == '__main__':
	print('10001st prime number is:', nth_prime(10001))
